File: JS/CodeWars/Storage/CW-VowelCount.js
/* 
CW-VowelCount
Return the number (count) of vowels in the given string.
We will consider a, e, i, o, and u as vowels for this Kata.
The input string will only consist of lower case letters and/or spaces.
 */

function getCount(str) {
  var vowelsCount = 0;
  var test = str.match(/[aeiou]/ig)
  if (test !== null){
    vowelsCount = test.length;
  }
  else{
    vowelsCount = 0;
  }
   return vowelsCount;
}

/* ############################################# OTHERS SOLUTIONS
2)
function getCount(str) {
  return (str.match(/[aeiou]/ig)||[]).length;
}
----------------------------------------------------------------------
3)
function getCount(str) {
  return str.replace(/[^aeiou]/gi, '').length;
}
----------------------------------------------------------------------
4)
function getCount(str) {
  let vowels = ['a','e','i','o','u'];
  return str.split('').filter(letter => {
    return vowels.includes(letter)? true : false;
  }).length;
}
----------------------------------------------------------------------
5)
function getCount(str) {
 return str.split('').filter(c => "aeiouAEIOU".includes(c)).length;
}
----------------------------------------------------------------------
6)
function getCount(str) {
  for (var n = 0, i = 0; i < str.length; i++) {
    var c = str.charCodeAt(i)
    if (c === 97 || c === 101 || c === 105 || c === 111 || c === 117) n++
  }

  return n
}
----------------------------------------------------------------------
function getCount(str) {
  var vowelsCount = 0;
  for (index in str){
    switch (str[index]) {
    case 'a':
    case 'e':
    case 'i':
    case 'o':
    case 'u':
    vowelsCount++;
    break;
    }
  }
  return vowelsCount;
}
----------------------------------------------------------------------

----------------------------------------------------------------------

----------------------------------------------------------------------

----------------------------------------------------------------------

----------------------------------------------------------------------

*/

File: Ruby/CodeWars/Collection Ruby 8 Kyu/CW-VowelCount.rb
=begin 
CW-VowelCount
Return the number (count) of vowels in the given string.
We will consider a, e, i, o, and u as vowels for this Kata.
The input string will only consist of lower case letters and/or spaces.
=end

def getCount( inputStr)
   inputStr.chars.select {|v| v=~(/[aeiou]/)}.count
end

puts getCount("murcielago")

=begin ############################################# OTHERS SOLUTIONS 
2)
def getCount(inputStr)
  arr = []
  inputStr.to_enum(:scan, /[aeiou]/).map{|x| arr << x}.length
 end
----------------------------------------------------------------------
3)
def getCount(inputStr)
 inputStr.downcase.count('aeiou')
end
----------------------------------------------------------------------
4)
def getCount(inputStr)
  inputStr.scan(/[aeiou]/i).size
end
----------------------------------------------------------------------
5)
def getCount(inputStr)
  inputStr.count 'aeiouAEIOU'
end
----------------------------------------------------------------------
6)
def getCount(str)
  str.size - str.tr('aeiou', '').size
end
----------------------------------------------------------------------
7)
def getCount(inputStr)
  vowels = %w{a e i o u}
  inputStr.split("").inject(0) do |acc, x|
    vowels.include?(x) ? acc + 1 : acc
  end
end
----------------------------------------------------------------------
8)
def getCount(inputStr)
  v = ['a', 'i', 'e', 'o', 'u']
  inputStr.split('').select {|l| v.include?(l)}.count
end
----------------------------------------------------------------------
9)
def getCount(inputStr)
  new_string = inputStr.gsub(/[aeiou]/, "*")
  new_string.count("*")
end/
----------------------------------------------------------------------

----------------------------------------------------------------------

=end